src/downloader.rs

```diff
@@ -63,10 +65,7 @@ fn download_url(base_url: &Url, version: &Version, arch: &Arch) -> Url {
     .unwrap()
 }
 
-pub fn extract_archive_into<P: AsRef<Path>>(
-    path: P,
-    response: crate::http::Response,
-) -> Result<(), Error> {
+fn extract_archive_into<P: AsRef<Path>, R: Read>(path: P, response: R) -> Result<(), Error> {
```

Comment:
<picture data-single-emoji=":interesting:" title=":interesting:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/interesting/0a282f287853ccd5.gif" alt=":interesting:" align="absmiddle"></picture>

Reply:
just a thought (not important, can be done in a different PR): what do you think about starting to use `impl` instead of generic constraints? seems more readable right?

```suggestion
fn extract_archive_into(path: impl AsRef<Path>, response: impl Read) -> Result<(), Error> {
```